### PR TITLE
Feature/eui 2871 create caseworker

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by paul-graham — CHANGES_REQUESTED** [review]
All the get methods in `caseWorkerService.ts` are essentially identical. The only difference is with what gets logged. It should be refactored to modularise everything _other_ than the logging or to have a single get method with a more generic log.

I'd favour the single get method as it already logs the path, which I would argue is sufficient distinction between the different endpoints.

File: api/test/pact/pact-tests/get-for-location-caseworker.spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { Pact } from '@pact-foundation/pact'
+import { assert } from 'chai'
+import * as getPort from 'get-port'
+import * as path from 'path'
+import { EnhancedRequest } from '../../../lib/models'
+import { handleCaseWorkerForLocation } from '../../../workAllocation/caseWorkerService'
+
+describe("Work Allocation for location Caseworker API", () => {
+
+    let MOCK_SERVER_PORT
```
Comment (ficklephil) [inline api/test/pact/pact-tests/get-for-location-caseworker.spec.ts:10]: Not sure Capitals should be used here, as this is not a static constant.

File: api/test/pact/pact-tests/get-caseworker-by-id.spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { Pact } from '@pact-foundation/pact'
+import { assert } from 'chai'
+import * as getPort from 'get-port'
+import * as path from 'path'
+import { EnhancedRequest } from '../../../lib/models'
+import { handleCaseWorkerDetails } from '../../../workAllocation/caseWorkerService'
+
+describe("Work Allocation for location and service Caseworker API", () => {
```
Comment (ficklephil) [inline api/test/pact/pact-tests/get-caseworker-by-id.spec.ts:8]: We're using both single quotes and double quotes within this file, let's just use single quotes for consistency.

File: api/test/pact/pact-tests/get-caseworker-by-id.spec.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    let MOCK_SERVER_PORT
+    let workallocationUrl
+    let provider
+    const mockResponse = {
+          "firstName": "firstName",
```
Comment (ficklephil) [inline api/test/pact/pact-tests/get-caseworker-by-id.spec.ts:14]: Double quotes on the name, not required here.

File: api/test/pact/pact-tests/get-caseworker-by-id.spec.ts (hunk 2 above)
Comment (paul-graham) [reply api/test/pact/pact-tests/get-caseworker-by-id.spec.ts:8]: Good spot!

**Comment by udaydenduluri33** [comment]
> All the get methods in `caseWorkerService.ts` are essentially identical. The only difference is with what gets logged. It should be refactored to modularise everything _other_ than the logging or to have a single get method with a more generic log.

We do not have the Swagger so I will leave it like this. Having said the point is noted. Agree in Principle. @paul-graham

File: api/test/pact/pact-tests/get-for-location-caseworker.spec.ts (hunk 1 above)
Comment (udaydenduluri33) [reply api/test/pact/pact-tests/get-for-location-caseworker.spec.ts:10]: This is the normal convention for all the PACT tests. Happy to change it. but that means 32 changes in 8 files. Are you sure about this?

File: api/test/pact/pact-tests/get-caseworker-by-id.spec.ts (hunk 2 above)
Comment (udaydenduluri33) [reply api/test/pact/pact-tests/get-caseworker-by-id.spec.ts]: Done

**Comment by paul-graham** [comment]
> We do not have the Swagger so I will leave it like this. Having said the point is noted. Agree in Principle. @paul-graham

Isn't that _more_ the reason we should make it super simple now? We only need to complicate matters if it turns out that the Swagger isn't as simple as it seems...?

File: api/test/pact/pact-tests/get-caseworker-by-id.spec.ts (hunk 3 above)
Comment (udaydenduluri33) [reply api/test/pact/pact-tests/get-caseworker-by-id.spec.ts]: Done

**Review by paul-graham — APPROVED** [review]
Approved, with an understanding that my comments will be taken into account during the integration phase and duplicate code _must_ be sorted.

**Review by connorpgpmcelroy — APPROVED** [review]
I can probably derive some of this for the location-data-service

File: api/test/pact/pact-tests/get-for-location-caseworker.spec.ts (hunk 1 above)
Comment (ficklephil) [reply api/test/pact/pact-tests/get-for-location-caseworker.spec.ts:10]: Let's change it as we go :) It's a development convention that static constants are in full capitals. But nothing else should be.

**Review by ficklephil — APPROVED** [review]

**Pull request merged**